src/link/Wasm.zig

```diff
@@ -45,6 +46,15 @@ pub const FnData = struct {
     /// Locations in the generated code where function indexes must be filled in.
     /// This must be kept ordered by offset.
     idx_refs: std.ArrayListUnmanaged(struct { offset: u32, decl: *Module.Decl }) = .{},
+    /// A list of `Inst` used to save and retrieve the indices of local variables
+    locals: std.ArrayListUnmanaged(*Inst) = .{},
+
+    /// Returns the index of a local given a pointer to an `Inst`
+    pub fn getLocalidx(self: *FnData, inst: *Inst) ?u32 {
+        return for (self.locals.items) |local, idx| {
+            if (local == inst) break @intCast(u32, idx);
+        } else null;
+    }
```

Comment:
Note that this will break when function arguments are implemented as they share the index space with locals. Might want to add a TODO.